plugins/modules/win_acl_inheritance.ps1

```diff
@@ -1,75 +1,156 @@
 #!powershell
 
+# Copyright: (c) 2022, Oleg Galushko (@inorangestylee)
 # Copyright: (c) 2015, Hans-Joachim Kliemeck <git@kliemeck.de>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
 
-$params = Parse-Args $args -supports_check_mode $true
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false
+$spec = @{
+    options = @{
+        path = @{ type = 'str'; required = $true }
```

Comment:
How does the `path` (ansible) type prevent it from working with registry keys? It still returns a string, with the vars already expanded, so I can't see why it would fail. Being able to expand env vars could in some circumstances be useful even with reg paths, but more importantly, it ensures that the the documentation is still clear.

If this does need to change type to `str`, let's also ensure the option's description notes that it still expands paths.

Reply:
it`s throw an exception at this line:
https://github.com/ansible/ansible/blob/e4890afc4e2af6c5126d056a2b4fdb1f1f122cf6/lib/ansible/module_utils/csharp/Ansible.Basic.cs#L594
```
Path.GetFullPath(stringValue);
```
Here is different behavior in PS5 and PS7. An exception is thrown if there is a ":" inside the string
```powershell
PS C:\Users\orange> $PsVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.22000.832
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22000.832
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKEY_LOCAL_MACHINE\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
Exception calling "GetFullPath" with "1" argument(s): "The given path's format is not supported."
At line:1 char:1
+ [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE:\SOFTWARE\AdminAbii ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : NotSupportedException

PS C:\Users\orange>
PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
Exception calling "GetFullPath" with "1" argument(s): "The given path's format is not supported."
At line:1 char:1
+ [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE:\SOFTWARE\AdminAbii ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : NotSupportedException

PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKEY_LOCAL_MACHINE\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKLM:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
Exception calling "GetFullPath" with "1" argument(s): "The given path's format is not supported."
At line:1 char:1
+ [System.IO.Path]::GetFullPath("HKLM:\SOFTWARE\AdminAbiit\win_acl_inhe ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : NotSupportedException

PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKLM\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKLM\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
```
and 
```powershell
PS C:\Users\orange> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKEY_LOCAL_MACHINE\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKEY_LOCAL_MACHINE:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKEY_LOCAL_MACHINE:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKLM:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKLM:\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
PS C:\Users\orange> [System.IO.Path]::GetFullPath("HKLM\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)")
C:\Users\orange\HKLM\SOFTWARE\AdminAbiit\win_acl_inheritance .ÅÑŚÌβŁÈ [$!@^&test(;)
```

Reply:
Yea sorry for not replying, if you have the type as `path` it will attempt to validate it as an actual file path which fails on registry paths (more than 1 char for the root).